### PR TITLE
Expose PID as metadata in CreateFile, OpenFile and FlushFile operations

### PR DESCRIPTION
**Pull request by dotslash**
This will help us with https://github.com/kahing/goofys/issues/273

### CODE REVIEW

File: fuseops/ops.go [hunk 1]
```diff
@@ -280,6 +286,9 @@ type CreateFileOp struct {
 	Name string
 	Mode os.FileMode
 
+	// Metadata
+	Metadata OpMetadata
```
Comment (kahing) [inline fuseops/ops.go]: move these to the beginning of the struct (so it's easier to see which operations have this and which don't)

File: samples/memfs/memfs_test.go [hunk 2]
```diff
@@ -696,6 +698,27 @@ func (t *MemFSTest) WriteOverlapsEndOfFile() {
 	ExpectEq("\x00\x00taco", string(contents))
 }
 
+func (t *MemFSTest) TestPid() {
+	// Touch the test file.
+	// Writes to files whose names start with "pid_test_" are locked to the
+	// process that created the file. Touching the file via exec.Command results
+	// in the test not being able to write to the file.
+	filename := path.Join(t.Dir, "pid_test_file")
+	fmt.Println(filename)
```
Comment (kahing) [inline samples/memfs/memfs_test.go]: remove debugging println

File: samples/memfs/memfs_test.go [hunk 3]
```diff
@@ -696,6 +698,27 @@ func (t *MemFSTest) WriteOverlapsEndOfFile() {
...
+	_, touchErr := touchCmd.Output()
+	AssertEq(nil, touchErr)
+
+	// Open the test file for appending.
+	f, err := os.OpenFile(filename, os.O_APPEND, 0600)
+	t.ToClose = append(t.ToClose, f)
+
+	AssertEq(nil, err)
+	// Write should fail.
+	_, err = f.Write([]byte("something"))
```
Comment (kahing) [inline samples/memfs/memfs_test.go]: add a test for open / flush as well

File: fuseops/ops.go (hunk 1 above)
Comment (dotslash) [reply fuseops/ops.go]: Done.

File: samples/memfs/memfs_test.go (hunk 2 above)
Comment (dotslash) [reply samples/memfs/memfs_test.go]: not relevant anymore.

File: samples/memfs/memfs_test.go (hunk 3 above)
Comment (dotslash) [reply samples/memfs/memfs_test.go]: discussed offline. Now we are testing  by verifying that memfs requests have non-zero pid

File: samples/memfs/memfs.go [hunk 4]
```diff
@@ -365,6 +365,10 @@ func (fs *memFS) createFile(
 func (fs *memFS) CreateFile(
 	ctx context.Context,
 	op *fuseops.CreateFileOp) (err error) {
+	if op.Metadata.Pid == 0 {
+		// CreateFileOp should have a valid pid in metadata.
+		err = fuse.EINVAL
```
Comment (kahing) [inline samples/memfs/memfs.go]: return right away without also doing the createFile

Comment (kahing) [reply samples/memfs/memfs.go]: same in the other ops

Comment (dotslash) [reply samples/memfs/memfs.go]: done

**Pull request merged**